scripts/pkg/build_templates/opensearch/deb/debian/preinst

```diff
@@ -23,6 +25,18 @@ if command -v systemctl >/dev/null && systemctl is-active opensearch-performance
     systemctl --no-reload stop opensearch-performance-analyzer.service
 fi
 
+# Check if OPENSEARCH_INITIAL_ADMIN_PASSWORD is defined
+# NOTE:
+# 1. This check will need to be modified if there will be a min dist for deb in future (currently there is none)
+# 2. Currently, the demo config setup is defined to run, in postinst, if `opensearch-security` is present. Cannot apply the same check here since the plugins folder is not available yet.
+OPENSEARCH_REQUIRED_VERSION="2.12.0"
+COMPARE_VERSION=`echo $OPENSEARCH_REQUIRED_VERSION $OPENSEARCH_VERSION | tr ' ' '\n' | sort -V | uniq | head -n 1`
+if [ "$COMPARE_VERSION" == "$OPENSEARCH_REQUIRED_VERSION" ] && [ -z "$OPENSEARCH_INITIAL_ADMIN_PASSWORD" ]; then
```

Comment:
1. Use `=` is enough for bash to compare. Nit.
2. So your logic is if version is not at least 2.12, and $OPENSEARCH_INITIAL_ADMIN_PASSWORD is none, fail. But what if the version is there but password not? It will not error out here. Probably using 
```
if [ "$COMPARE_VERSION" != "$OPENSEARCH_REQUIRED_VERSION" ] || [ -z "$OPENSEARCH_INITIAL_ADMIN_PASSWORD" ]; then

```

Please correct me if I am wrong here.

Reply:
we will have to keep == otherwise case when version is defined will fail. However to address the issue you bring, we can start the check with `if [ $OPENSEARCH_VERSION != CHANGE_VERSION ] && `

Reply:
and added another check to skip checking the environment variable if OpenSearch is already installed

Reply:
> we will have to keep == otherwise case when version is defined will fail. However to address the issue you bring, we can start the check with `if [ $OPENSEARCH_VERSION != CHANGE_VERSION ] && `

In bash `=` and `==` in if condition is the same.

Reply:
To complete @peterzhuamazon answer, `[ foo = bar ]` is POSIX correct. `[ foo == bar ]` is a bashism and should be avoided.